Disable Cloudwatch Log-Group creation by “insights-results-aggregator”

Disable Cloudwatch Log-Group creation by “insights-results-aggregator”?  The log group is now created by the `app-sre automation`, which blocks `ccs-data-pipeline` deployment when it tries to create the log group, which has already been created by app-sre.

For ccx-data-pipeline, written in Python, I did:
```
    handler = CloudWatchLogHandler(
        boto3_session=session,
        log_group=os.environ["CW_LOG_GROUP"],
        stream_name=os.environ["CW_STREAM_NAME"],
        create_log_group=False,
    )
```

Rolling out `insights-results-aggregator` produced the following in the pod log:
```
aarif@MacBook-Pro~... $ oc logs -f insights-results-aggregator-4-zss7m
panic: AccessDeniedException: User: arn:aws:iam::351356922033:user/ccx-data-pipeline-stage is not authorized to perform: logs:CreateLogStream on resource: arn:aws:logs:us-east-1:351356922033:log-group:platform-dev:log-stream:insights-results-aggregator
	status code: 400, request id: dbf4cd63-0039-4cf3-834a-97db5cf2d108

goroutine 1 [running]:
main.main()
	/opt/app-root/src/aggregator.go:439 +0x1d7
aarif@MacBook-Pro~... $ 
```

--- a/config-devel.toml
+++ b/config-devel.toml
@@ -47,3 +47,4 @@ aws_region = "us-east-1"
 log_group = "platform-dev"
 stream_name = "insights-results-aggregator"
 debug = false
+create_stream_if_not_exists = true

--- a/config.toml
+++ b/config.toml
@@ -42,3 +42,4 @@ aws_region = "us-east-1"
 log_group = "platform-dev"
 stream_name = "insights-results-aggregator"
 debug = false
+create_stream_if_not_exists = false

--- a/logger/configuration.go
+++ b/logger/configuration.go
@@ -38,12 +38,13 @@ type LoggingConfiguration struct {
 
 // CloudWatchConfiguration represents configuration of CloudWatch logger
 type CloudWatchConfiguration struct {
-	AWSAccessID     string `mapstructure:"aws_access_id" toml:"aws_access_id"`
-	AWSSecretKey    string `mapstructure:"aws_secret_key" toml:"aws_secret_key"`
-	AWSSessionToken string `mapstructure:"aws_session_token" toml:"aws_session_token"`
-	AWSRegion       string `mapstructure:"aws_region" toml:"aws_region"`
-	LogGroup        string `mapstructure:"log_group" toml:"log_group"`
-	StreamName      string `mapstructure:"stream_name" toml:"stream_name"`
+	AWSAccessID             string `mapstructure:"aws_access_id" toml:"aws_access_id"`
+	AWSSecretKey            string `mapstructure:"aws_secret_key" toml:"aws_secret_key"`
+	AWSSessionToken         string `mapstructure:"aws_session_token" toml:"aws_session_token"`
+	AWSRegion               string `mapstructure:"aws_region" toml:"aws_region"`
+	LogGroup                string `mapstructure:"log_group" toml:"log_group"`
+	StreamName              string `mapstructure:"stream_name" toml:"stream_name"`
+	CreateStreamIfNotExists bool   `mapstructure:"create_stream_if_not_exists" toml:"create_stream_if_not_exists"`
 
 	// enable debug logs for debugging aws client itself
 	Debug bool `mapstructure:"debug" toml:"debug"`

--- a/logger/logger.go
+++ b/logger/logger.go
@@ -118,11 +118,19 @@ func InitZerolog(loggingConf LoggingConfiguration, cloudWatchConf CloudWatchConf
 			WithLogLevel(awsLogLevel)
 
 		cloudWatchSession := session.Must(session.NewSession(awsConf))
-		group := cloudwatch.NewGroup(cloudWatchConf.LogGroup, cloudwatchlogs.New(cloudWatchSession))
+		cloudWatchClient := cloudwatchlogs.New(cloudWatchSession)
 
-		cloudWatchWriter, err := group.Create(cloudWatchConf.StreamName)
-		if err != nil {
-			return err
+		var cloudWatchWriter io.Writer
+		if cloudWatchConf.CreateStreamIfNotExists {
+			group := cloudwatch.NewGroup(cloudWatchConf.LogGroup, cloudWatchClient)
+
+			var err error
+			cloudWatchWriter, err = group.Create(cloudWatchConf.StreamName)
+			if err != nil {
+				return err
+			}
+		} else {
+			cloudWatchWriter = cloudwatch.NewWriter(cloudWatchConf.LogGroup, cloudWatchConf.StreamName, cloudWatchClient)
 		}
 
 		writers = append(writers, &UnJSONWriter{Writer: cloudWatchWriter})

